org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/util/GitCredentialsService.java

```diff
@@ -0,0 +1,355 @@
+	 *            Store, if they already exist in the Secure Store.
+	 */
+	public static void deleteCredentialsFromSecureStorage(String repositoryURL) {
+		ISecurePreferences preferences = SecurePreferencesFactory.getDefault();
+		String hashedURL = getMD5HashForUrl(repositoryURL);
+		if (hashedURL != null && preferences.nodeExists(hashedURL)) {
+			ISecurePreferences node = preferences.node(hashedURL);
+			node.removeNode();
+		}
+	}
+
+	/**
+	 *
+	 * @param repositoryURL
+	 * @return the credentials from Secure Store for the given repository url
+	 */
+
+	private static IExternalRepositoryInfoRequest getRepoCredentialsFromSecureStorage(String repositoryURL) {
+		ISecurePreferences preferences = SecurePreferencesFactory.getDefault();
+		String hashedURL = getMD5HashForUrl(repositoryURL);
+		if (hashedURL != null && preferences.nodeExists(hashedURL)) {
+			ISecurePreferences node = preferences.node(hashedURL);
+			return new IExternalRepositoryInfoRequest() {
+				@Override
+				public String getUser() {
+					try {
+						return node.get("user", null); //$NON-NLS-1$
+					} catch (StorageException e) {
+						AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+					}
+					return null;
+				}
+
+				@Override
+				public String getUrl() {
+					return repositoryURL;
+				}
+
+				@Override
+				public String getPassword() {
+					try {
+						return node.get("password", null); //$NON-NLS-1$
+					} catch (StorageException e) {
+						AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+					}
+
+					return null;
+				}
+			};
+		}
+
+		return null;
+	}
+
+	/**
+	 * Returns MD5 hash value of the given repository URL. Used as a node value
+	 * in the secure storage, due limitations of using actual repository url as
+	 * node value in secure storage. Secure storage limits the length of node
+	 * value and also limits the use of special characters.
+	 *
+	 * @param url
+	 * @return MD5 hash value of the URL
+	 */
+	public static String getMD5HashForUrl(String url) {
+		String hashedURL = null;
+
+		if (url != null) {
+			try {
+				MessageDigest messageDigest = MessageDigest.getInstance("MD5"); //$NON-NLS-1$
+				messageDigest.update(url.getBytes());
+				byte[] digest = messageDigest.digest();
+				hashedURL = DatatypeConverter.printHexBinary(digest).toUpperCase();
+			} catch (NoSuchAlgorithmException e) {
+				AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+			}
+		}
+		return hashedURL;
+
+	}
+
+	/**
+	 * @param dialog
+	 *            Display longText in the DialogTray of the given dialog
+	 */
+	public static void handleExceptionAndDisplayInDialogTray(Exception e, TrayDialog dialog) {
+		if (e instanceof ResourceException && GitCredentialsService.isAuthenticationIssue((ResourceException) e)) {
```

Comment:
Can we have this method as a generic error handler and not specifically for authorization related issues?
I mean we do not need the condition check right?

Reply:
Removed the check for AuthorizationIssue.